MonkeyBot/Modules/Polls/PollMessageUpdater.cs

```diff
@@ -0,0 +1,92 @@
+﻿using DSharpPlus;
+using DSharpPlus.CommandsNext;
+using DSharpPlus.Entities;
+using MonkeyBot.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace MonkeyBot.Modules
+{
+    public class PollMessageUpdater
+    {
+        private DiscordMessage _pollMessage;
+        private DiscordMessageBuilder _pollPreviewMessage;
+        private DiscordEmbedBuilder _pollPreviewMessageEmbed;
+
+        private PollMessageUpdater(CommandContext ctx, DiscordEmbedBuilder pollPreviewMessageEmbed,
+            DiscordMessage pollMessage)
+        {
+            _pollMessage = pollMessage;
+            _pollPreviewMessageEmbed = pollPreviewMessageEmbed;
+
+            _pollPreviewMessage = new DiscordMessageBuilder()
+                .WithEmbed(_pollPreviewMessageEmbed.Build());
+        }
+
+        public DiscordMessage Message => _pollMessage;
+
+        public static string BuildDescription(IEnumerable<PollAnswer> pollAnswers) =>
+            string.Join("\n",
+                pollAnswers.Select(ans => $"{ans.Emoji}`[{ans.Count}]` {ans.Value}"));
+
+        public static IEnumerable<DiscordActionRowComponent>
+            BuildAnswerButtons(IEnumerable<PollAnswer> pollAnswers) =>
+            pollAnswers
+                .GroupBy(x => x.OrderNumber / 5)
+                .Select(g => new DiscordActionRowComponent(g.Select(x => new DiscordButtonComponent(
+                    ButtonStyle.Primary, x.Id, string.Empty,
+                    emoji: new DiscordComponentEmoji(x.Emoji)))));
+        
+        public static async Task<PollMessageUpdater> Create(CommandContext ctx)
+        {
+            var pollEmbedBuilder = new DiscordEmbedBuilder()
+                .WithTitle("New poll")
+                .WithColor(DiscordColor.Azure)
+                .WithDescription("...")
+                .WithAuthor(ctx.Member.Username, iconUrl: ctx.Member.AvatarUrl);
+            var pollMessage = await ctx.RespondAsync(pollEmbedBuilder.Build());
+            return new PollMessageUpdater(ctx, pollEmbedBuilder, pollMessage);
+        }
+
+
+        public async Task SetPollTitle(string title) => await WithTitle($"**Poll: {title}**");
+
+        public async Task SetEndTime(DateTime endTime) => await WithFooter($"Poll will end on {endTime:dd.MM.yyyy} at {endTime:HH:mm \"UTC\"zz}");
```

Comment:
This could use the new [Discord time formatting](https://www.reddit.com/r/discordapp/comments/ob2h2l/discord_added_new_timestamp_formatting/) so it will be displayed in the users time zone. I think it is `Formatter.Timestamp()`

Reply:
I have tried to use that Formatter.Timestamp, but Discord isn't converting <t:TIMESTAMP:FLAG> when it is in embed footer. 
![image](https://user-images.githubusercontent.com/22114978/139557221-c8ebbc70-64cf-4564-adca-fe0f23d8e625.png)

Reply:
Ok, I will look into it at some point if its a DSharp issue or not possible for embed footers